pshtt/pshtt.py

```diff
@@ -594,14 +605,34 @@ def https_check(endpoint):
 
     try:
         cert_plugin_result = None
-        command = sslyze.plugins.certificate_info_plugin.CertificateInfoScanCommand(ca_file=CA_FILE)
-        scanner = sslyze.synchronous_scanner.SynchronousScanner()
-        cert_plugin_result = scanner.run_scan_command(server_info, command)
+        scanner = Scanner()
+        command = ScanCommand.CERTIFICATE_INFO
+        if CA_FILE is not None:
+            command_extra_args = {
+                ScanCommand.CERTIFICATE_INFO: CertificateInfoExtraArguments(custom_ca_file=Path(CA_FILE))
+            }
+            scan_request = ServerScanRequest(
+                server_info=server_info,
+                scan_commands_extra_arguments=command_extra_args,
+                scan_commands=[command]
+            )
+        else:
+            scan_request = ServerScanRequest(
+                server_info=server_info,
+                scan_commands=[command]
+            )
+        scanner.queue_scan(scan_request)
+        # Retrieve results from generator object
+        scan_result = [x for x in scanner.get_results()][0]
+        cert_plugin_result = scan_result.scan_commands_results[ScanCommand.CERTIFICATE_INFO]
     except Exception as err:
         try:
             if "timed out" in str(err):
                 logging.exception("{}: Retrying sslyze scanner certificate plugin.".format(endpoint.url))
-                cert_plugin_result = scanner.run_scan_command(server_info, command)
+                scanner.queue_scan(scan_request)
+                # Retrieve results from generator object
```

Comment:
Just wanted to highlight that we're consume the iterator, and we are only using the first result. This should be fine for how this library does scanning, but I still think it's worth explicitly commenting.

```suggestion
                # Consume the generator object and retrieve the first result
```